clients/src/main/java/org/apache/kafka/clients/producer/MockProducer.java

```diff
@@ -172,6 +176,15 @@ public void sendOffsetsToTransaction(Map<TopicPartition, OffsetAndMetadata> offs
         this.sentOffsets = true;
     }
 
+    @Override
+    public void sendOffsetsToTransaction(Map<TopicPartition, OffsetAndMetadata> offsets,
+                                         ConsumerGroupMetadata groupMetadata) throws ProducerFencedException,
+                                                                                     IllegalGenerationException,
+                                                                                     UnknownMemberIdException,
```

Comment:
nit: this definition looks really awkward

Reply:
You mean exception throwing right?

Reply:
Yeah, since these are runtime exceptions, I think you can leave them off. Users can see the documentation to see what errors we throw.